From 0abd9fc5212a6c6d17fb0c8c40c77a83aee6bc26 Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Tue, 5 Nov 2024 12:19:59 +0530
Subject: [PATCH] pcie: qcom: Add device ID for IPQ5424 target

Adding device ID check in get link width/speed
API's for IPQ5424 target.

Change-Id: I88b7dec9ebae64dbc5d04b28c8af4efa3ae4b169
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index 172c9f22631d..35d69d652977 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -208,6 +208,7 @@
 #define PCIE_CAP_TARGET_LINK_SPD_MASK		GENMASK(3, 0)
 #define QCOM_IPQ9574_DEVICE_ID			0x1108
 #define QCOM_IPQ5332_DEVICE_ID			0x1005
+#define QCOM_IPQ5424_DEVICE_ID			0x1006
 #define PCIE20_LNK_CONTROL2_LINK_STATUS2	0xa0
 
 #define QCOM_PCIE_1_0_0_MAX_CLOCKS		4
@@ -1338,7 +1339,9 @@ int pcie_set_link_speed(struct pci_dev *dev, u16 target_link_speed)
 	struct dw_pcie *pci;
 	u32 val;
 
-	if (dev->device != QCOM_IPQ9574_DEVICE_ID && dev->device != QCOM_IPQ5332_DEVICE_ID)
+	if (dev->device != QCOM_IPQ9574_DEVICE_ID &&
+	    dev->device != QCOM_IPQ5332_DEVICE_ID &&
+	    dev->device != QCOM_IPQ5424_DEVICE_ID)
 		return -EINVAL;
 
 	if (target_link_speed < 1 || target_link_speed > 3)
@@ -1378,7 +1381,9 @@ int pcie_set_link_width(struct pci_dev *dev, u16 target_link_width)
 	struct qcom_pcie *pcie;
 	u32 val;
 
-	if (dev->device != QCOM_IPQ9574_DEVICE_ID && dev->device != QCOM_IPQ5332_DEVICE_ID)
+	if (dev->device != QCOM_IPQ9574_DEVICE_ID &&
+	    dev->device != QCOM_IPQ5332_DEVICE_ID &&
+	    dev->device != QCOM_IPQ5424_DEVICE_ID)
 		return -EINVAL;
 
 	if (target_link_width < 1 || target_link_width > 2)
-- 
2.34.1

